onto_spread_ed/release/GenerateHierarchicalSpreadsheetReleaseStep.py

```
@@ -0,0 +1,180 @@
+    def height(self) -> int:
+        return max((c.height() for c in self.children), default=0) + 1
+
+    def recurse(self, fn: Callable[[Self], Any]):
+        fn(self)
+        for c in self.children:
+            c.recurse(fn)
+
+
+def form_tree(edges: List[Tuple[Tuple[str, str, str], Optional[str]]]) -> List[Node]:
+    all_nodes = set(n for n, _ in edges)
+    item_to_node = dict((c, Node(item=c, label=l if l is not None else c, definition=d)) for (c, l, d) in all_nodes)
```

Ruff

Ambiguous variable name: `l`

(E741)